lib/queue.js

```diff
@@ -16,6 +16,29 @@ var allowedStatistics = [
     "actualDataSize"
 ];
 
+function noop() {}
+
+/**
+ * Recursive function that compares immutable properties from the first object
+ * to the same properties in the second one and returns true if all of the
+ * properties matched. If not, it returns false.
+ */
+function compare (a, b) {
+    var matches;
+
+    for (var key in a) {
+        if (a.hasOwnProperty(key)) {
+            if (typeof a[key] === "object" && typeof b[key] === "object") {
+                matches = compare(a[key], b[key]);
+            } else {
+                matches = matches === false ? matches : a[key] === b[key];
```

Comment:
I think we could bail out early here. If any of the matches come back false, we should immediately return false rather than continuing to loop. :)